atlas/templates/Jenkinsfile.gotmpl

```diff
@@ -31,21 +37,24 @@ pipeline {
       steps {
         withDockerRegistry([credentialsId: "<insert-the-creds-id>", url: ""]) {
           dir("$DIRECTORY") {
-            sh "make docker push"
+            // The shell started by Jenkins should inherit the env vars defined above.
+            // Since Makefile.common uses ?= to set GIT_COMMIT and IMAGE_VERSION, then the values
+            // set above should override the defaults from Makefile.common.
+            sh "make docker"
           }
         }
       }
     }
     stage("Push") {
       when {
-        branch "master"
+        // This check will work regardless of whether the merge build runs off of the master
+        // or develop branches.
+        expression{ !isPrBuild() }
       }
       steps {
-        withDockerRegistry([credentialsId: "<insert-the-creds-id>", url: ""]) {
-          dir("$DIRECTORY") {
-            sh "make push IMAGE_VERSION=latest"
-          }
-        }
+        // Just guessing here that {{.Name}} will be the docker image name (sans tag).
+        // FYI - Due to immutable tags in Harbor, the 'latest' tag cannot be used.
+        signDockerImage({{ .Name }}, env.IMAGE_VERSION)
```

Comment:
So the previous statement is using a well known Jenkins plugin that anybody in open source can use. Are we planning to open source the signDockerImage library? If not, we need to sort out a different pattern here. The toolkit is open source, so we need to handle how to layer on proprietary logic here.

Reply:
Another issue here, how do developers test this locally without Jenkins? If a Jenkins job fails, it should be possible to replay the actions it took locally to remediate the problem. The Jenkinsfile is full of make calls for that reason.

Reply:
OK - I wasn't aware of that.
Is there an open-source use-case for contributing code that we run in GovCloud.  We're not going to open-source image signing. Maybe we templatize this part of the Jenkinsfile such that bloxers can generate a Jenkinsfile that has the signing step, and outside collaborators generate a Jenkinsfile with a simple `make push`?

Reply:
re: debugging, the signing process requires a delegate key along with a whole lot of other info.  From a security point of view, we wouldn't want to share all of those creds with a wide audience, so I think we'd have to look at that on a case-by-case basis - or perhaps allow each developer to define/register their own delegate key pair.

Reply:
Sounds like we need a non-signing process for harbor in addition to the signed one. It's a very common workflow to push your own copy of an application to one of the dev clusters. If they can't push to harbor, they'll need something else.

Reply:
> The toolkit is open source, so we need to handle how to layer on proprietary logic here.

Since this is a go template, conditional statements are available. These are used for many other optional features, so there are ample examples elsewhere in this repo of ways to make this step optional.